Documentation/MigrationGuide.md

```diff
@@ -1,15 +1,116 @@
+
+## Commands can have multiple outputs
+
+This is only available for users with a configuration file, not via command line invocation. For each command you can now have a list of `outputs`, each output with a template name (`templateName`) or path (`templatePath`), and a list of parameters (`params`). For example:
+
+```yaml
+ib:
+  inputs: dir/to/search/for/storyboards
+  outputs:
+    - templateName: scenes-swift4
+      output: Storyboard Scenes.swift
+    - templateName: segues-swift4
+      output: Storyboard Segues.swift
+```
+
+As an extra advantage, the resources for that command will only be parsed once, giving you a nice performance boost.
+
+You'll notice that the configuration structure has changed a little bit, see below for more information.
+
+## Configuration changes
+
+Now that each command can have multiple outputs, we're deprecating the following configuration settings.
+
+- `output` has been renamed `outputs`, which now accepts one or more items.
+- `templateName`, `templatePath` and `params` have been moved down to the `outputs` level, so that each output can have a template name (or path) and parameters.
+- `paths` has been renamed to `inputs`.
+
+So if you had the following configuration:
+
+```yaml
+colors:
+  paths: path/to/colors.json
+  templateName: swift4
+  output: Colors.swift
+storyboards:
+  paths: path/to/storyboards
+  templateName: swift4
+  output: Storyboards.swift
+```
+
+It would become:
+
+```yaml
+colors:
+  inputs: path/to/colors.json
+  outputs:
+  	templateName: swift4
+    output: Colors.swift
+ib:
+  inputs: path/to/storyboards
+  outputs:
+    - templateName: scenes-swift4
+      output: Storyboard Scenes.swift
+    - templateName: segues-swift4
+      output: Storyboard Segues.swift
+```
+
+## New Features
+
+We've added many new features to SwiftGen since our last release. For example, you can now install SwiftGen using Mint, see our [installation instructions](../README.md#installation) for more information.
+
+### JSON, Plist and YAML support
+
+You can read all about it in our [Read Me](../README.md). SwiftGen can now parse JSON, Plist and YAML files. We even provide a few bundled templates for each of these commands to get you started, so you can access your data in a type-safe and easy way:
+
+* `inline-swift3/4`: These templates embed the contents of the data into the swift file, so it doesn't have to be loaded at runtime.
+* `runtime-swift3/4`: With these templates, you can generate swift code that will load the underlying JSON/Plist file at runtime, and parses the content in a type-safe way.
+
+Please note that the bundled templates are only meant for the most basic uses for these data types. If you need to access these data files in a different way, using a different structure, or for a specific purpose, you may want to write your own templates. For more information, read the [creating your own templates](Creating-your-templates.md) documentation.
```

Comment:
> […] in a different way, especially since you'll probably use some custom structure in those JSON/Plist/YAML files you use, you may want to write your own templates that better fits those structures (for example to write a template that expects specific keys in a YAML to generate code based on the values in those keys…).
>
> For more information […]